### PR TITLE
Core/Organization refactoring

### PR DESCRIPTION
**Pull request by dbalintx**
### Feature or Bugfix
- Refactoring

### Detail
Adding service layer to Organizations in order to follow the resolver-service-repository structure we established for v2.0 modularization.
Also cleanup some dead code.

### Relates
[#741 ](https://github.com/data-dot-all/dataall/issues/741)

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dbalintx** [comment]
Testing:

- [x] unit tests

Local deployment
- [x] Create organization
- [x] Modify organization
- [x] Get organization
- [x] List organizations
- [x] Invite group to organization
- [x] List organization groups
- [x] Remove group from organization
- [ ] Link environment to organization
- [ ] List organization environments
- [x] Archive organization

AWS deployment
- [ ] Create organization
- [ ] Modify organization
- [ ] Get organization
- [ ] List organizations
- [ ] Invite group to organization
- [ ] List organization groups
- [ ] Remove group from organization
- [ ] Link environment to organization
- [ ] List organization environments
- [ ] Archive organization

File: backend/dataall/core/organizations/services/organization_service.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                session=session,
+                group=organization.SamlGroupName,
+                permissions=permissions.ORGANIZATION_ALL,
+                resource_uri=organization.organizationUri,
+                resource_type=models.Organization.__name__,
+            )
+            return organization
+
+    @staticmethod
+    # TODO: resource permission check for permissions.GET_ORGANIZATION ?
```
Comment (noah-paige) [inline backend/dataall/core/organizations/services/organization_service.py:88]: Let's add this resource permission check for `GET_ORGANIZATION`

Comment (noah-paige) [reply backend/dataall/core/organizations/services/organization_service.py:88]: > Let's add this resource permission check for `GET_ORGANIZATION`


Tested in AWS Deploymnet (after addressing `TODO` comment and adding `GET_ORGANIZATION` resource permission check of `def get_organization(...)`
- [ ] Create organization
- [ ] Modify organization
- [ ] Get organization
- [ ] List organizations
- [ ] Invite group to organization
- [ ] List organization groups
- [ ] Remove group from organization
- [ ] Link environment to organization
- [ ] List organization environments
- [ ] Archive organization

Think we can add this line then good to approve on my end!

Comment (dbalintx) [reply backend/dataall/core/organizations/services/organization_service.py]: Thanks a lot for deploying it! I added the permission check

**Review by noah-paige — APPROVED** [review]
looks good to me - approving

**Pull request merged**